packages/pulumi-aws/src/env/base.ts

```diff
@@ -0,0 +1,33 @@
+export interface ICreateGetEnvParams<T = unknown> {
+    name: string;
+    defaultValue: T;
+}
+
+/**
+ * TODO @adrian @pavel verify please
+ *
+ *
+ * We do not allow env with name "undefined" or "null". It is considered as an empty string.
+ * Sometimes, when setting env variable, a user error might set it to "undefined" or "null" value by mistake.
```

Comment:
Sounds good.... hopefully nobody will need these specific values in their env vars, but we'll see. :)